### PR TITLE
feat(sdk): Add method to get homeserver capabilities

### PR DESCRIPTION
**Pull request by zecakeh**
Useful to know if we can change the user password, among other things.

### CODE REVIEW

**Comment by zecakeh** [comment]
Done

File: crates/matrix-sdk/src/client.rs [hunk 1]
```diff
@@ -346,6 +347,33 @@ impl Client {
...
+    ///
+    /// This method should be used to check what features are supported by the
+    /// homeserver.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # use futures::executor::block_on;
+    /// # use matrix_sdk::Client;
+    /// # use url::Url;
+    /// # let homeserver = Url::parse("http://example.com").unwrap();
```
Comment (poljar) [inline crates/matrix-sdk/src/client.rs:360]: No need to unwrap.

**Review by poljar — CHANGES_REQUESTED** [review]
Please no new unwraps in doc examples. Otherwise looks good.

**Comment by zecakeh** [comment]
Unwrap removed

**Review by gnunicorn — APPROVED** [review]

**Pull request merged**